cnd/src/btsieve/block_by_hash.rs

```diff
@@ -0,0 +1,44 @@
+macro_rules! impl_block_by_hash {
+    () => {
+        impl<C> BlockByHash for Cache<C>
+        where
+            C: BlockByHash<Block = Block, BlockHash = Hash> + Clone,
+        {
+            type Block = Block;
+            type BlockHash = Hash;
+
+            fn block_by_hash(
+                &self,
+                block_hash: Self::BlockHash,
+            ) -> Box<dyn Future<Item = Self::Block, Error = anyhow::Error> + Send + 'static> {
+                let connector = self.connector.clone();
+                let cache = Arc::clone(&self.block_cache);
+                Box::new(Box::pin(block_by_hash(connector, cache, block_hash)).compat())
+            }
+        }
+
+        async fn block_by_hash<C>(
+            connector: C,
+            cache: Arc<Mutex<LruCache<Hash, Block>>>,
```

Comment:
These types should be fully qualified here to not have ambiguous imports that lead to weird compile errors.

Reply:
I'm getting the feeling you don't like how terse I made everything :) I added no-mergify back in.  Lets chat about it in person, easier to explain the nuances then.